tools/ipa_packageset_builder.py

```diff
@@ -0,0 +1,82 @@
+from jinja2 import Template
+import argparse
+from os import path
+from os.path import isfile
+DIR_PATH_TO_TEMPATES = "languages"
+DIR_PATH_TO_PACKAGES = "libraries"
+RUNNER_NAME = "Dockerfile.runner.j2"
+LANGSERVER_VERSION = "ce3f89098fddfaf4db8639a97ce3c0317abbd971"
+LANGSERVER_IMAGE ="algorithmiahq/langserver:{}".format(LANGSERVER_VERSION)
+RUNNER_PATH = path.join(DIR_PATH_TO_TEMPATES, RUNNER_NAME)
+
+class Package:
+    def __init__(self, package_name, install_script, dockerfile_path, base_image):
+        self.script = install_script
+        if dockerfile_path:
+            self.dockerfile = get_dockerfile_as_string(dockerfile_path)
+        else:
+            raise Exception('dockerfile path not available for package {}'.format(package_name))
+        if base_image:
+            self.base_image = base_image
+        else:
+            raise Exception('base_image path not available for package {}'.format(package_name))
+
+
+
+def get_dockerfile_as_string(file_path):
+    with open(file_path, 'r') as fileobject:
+        stringified = fileobject.read()
+    output = stringified.split('\n')
+    return output
+
+def get_template(template_path):
+    with open(template_path, 'r') as fileobject:
+        template_string = fileobject.read()
+    template = Template(template_string)
+    return template
+
+def save_generated_template(template, output_path):
+    with open(output_path, 'w') as fileobject:
+        fileobject.write(template)
+    return output_path
+
+def check_if_exists(filepath):
+    if isfile(filepath):
+        return filepath
+    else:
+        return None
+
+def build(base_image, package_dirs, output_file_path):
+
+    raw_template = get_template(RUNNER_PATH)
+    packages = []
+    for dir in package_dirs:
+        dockerfile_path = path.join('',  *[DIR_PATH_TO_PACKAGES, dir, "Dockerfile"])
```

Comment:
nit: can't you just do `path.join('', DIR_PATH_TO_PACKAGSE, dir, "Dockerfile")`?

Reply:
Does that work? I didn't think you could do that with path.join, I'll give er a shot